doc/common/glossary.md

```diff
@@ -20,94 +20,136 @@ To help guide the reader, this glossary provides an introduction to the terminol
+    >_*Note:*_ Can be used to specify VM Instance or Container Instance.
+
+- **Network Function (NF):**  functional block or application within a network infrastructure that has well-defined external interfaces and well-defined functional behaviour.
+  - Within **NFV**, A **Network Function** is implemented in a form of **Virtualised NF** or a **Cloud Native NF**.
+
+
+- **Network Function Virtualisation (NFV):** The concept of separating network functions from the hardware they run on by using a virtual hardware abstraction layer.
+
+- **Network Function Virtualisation Infrastructure (NFVI):** The totality of all hardware and software components used to build the environment in which a set of VAs are deployed (also referred to as cloud infrastructure).
+ >_*Note:*_ The NFVI can span across many locations, e.g. places where data centres or edge nodes are operated. The network providing connectivity between these locations is regarded to be part of the cloud infrastructure. **NFVI** and **VNF** are the top-level conceptual entities in the scope of Network Function Virtualisation. All other components are sub-entities of these two main entities.
+
+- **Network Service (NS):** composition of **Network Function**(s) and/or **Network Service**(s), defined by its functional and behavioural specification, including the service lifecycle.
+
+- **Virtual Application (VA):** A general term for software which can be loaded into a Virtual Machine.
+  >_*Note:*_ a **VNF** is one type of VA.
+
+- **Virtual Machine (VM):** virtualised computation environment that behaves like a physical computer/server.
+  >_*Note:*_ A **VM** consists of all of the components (processor (CPU), memory, storage, interfaces/ports, etc.) of a physical computer/server. It is created using Instance Type together with sizing information or Compute Flavour.  
+
+- **Virtual Machine instances Catalogue:** Pre-defined instance types and compute flavours.  
+
+- **Virtual Network Function (VNF):** a software implementation of a **Network Function**, capable of running on the **Cloud Infrastructure**.
+  - **VNF**s are built from one or more VNF Components (**VNFC**) and, in most cases,  the VNFC is hosted on a single VM or Container.
+
+
+- **Virtual resources:**
+  - **Virtual Compute resource (a.k.a. virtualisation container):** partition of a compute node that provides an isolated virtualised computation environment.
+  - **Virtual Storage resource:** virtualised non-volatile storage allocated to a virtualised computation environment hosting a **VNFC**
+  - **Virtual Networking resource:** routes information among the network interfaces of a virtual compute resource and physical network interfaces, providing the necessary connectivity
+
+
+- **Workload:** Workload refers to software running on top of compute resources such as **VMs** or **Containers**. Most relevant workload categories in context of cloud infrastructure are:
```

Comment:
I think we are using workload not for any sw running on compute resources but specifically to applications such as VAs, VNFs and CNFs. If yes then we need to modify the definition.

Reply:
Can I suggest this is handled in a PR dedicated to reviewing the definitions and cleaning the wording - this one seems quite specific to changing the order.

Reply:
We have opened #1906 to discuss this change.

Reply:
@pgoyal01, can we resolve this thread?